File: Tests/executors/CXX/core/src/DownloadExecutor.cpp
#include "DownloadExecutor.h"

void DownloadExecutor::doGet(framework::HttpRequest& request, framework::HttpResponse& response)
{
	const std::string& fileName = request.getJson().get<std::string>("fileName");

	request.streamFile(fileName, response, fileName);
}

DEFINE_EXECUTOR(DownloadExecutor)

File: Tests/executors/CSharp/CoreExecutors/DownloadExecutor.cs
using Framework;

public class DownloadExecutor : HeavyOperationStatelessExecutor
{
	public override void DoGet(HttpRequest request, HttpResponse response)
	{
		string fileName = request.GetJson()["fileName"].GetString()!;

		request.StreamFile(fileName, response, fileName);
	}
}
